components/concept/fetch/src/main/java/mozilla/components/concept/fetch/Client.kt

```diff
@@ -62,3 +66,31 @@ abstract class Client {
         "Connection" to "keep-alive"
     )
 }
+
+/**
+ * "Fetches" a base64 encoded data URI.
+ *
+ * @param request The [Request] for the data URI.
+ * @return The generated [Response] including the decoded bytes as body.
+ */
+@Suppress("TooGenericExceptionCaught")
+fun fetchDataUri(request: Request): Response {
```

Comment:
Should we put this into `Client` and make it `protected` (like `defaultHeaders`) so that it is not used outside of a `Client` implementation? This could now end up getting called by app code.

Reply:
OK, makes sense. Updated!